### DISCUSSION TITLE
GPT researcher unable to run on local document

### DISCUSSION THREAD

**Discussion by sankarsyneoshealth** [assafelovic/gpt-researcher#1005] [Q&A]
i am trying to run gpt-researcher in the local document but it is fetching the result from web. need solution to fix the issue.

**Comment by lmontin** [comment]
fixed here https://github.com/assafelovic/gpt-researcher/pull/1070

**Comment by belviskhoremk** [comment]
I am currently using the pip version  0.10.11 of gpt-researcher and was facing the same issue.

Instead of passing the value while initializing the class like this
researcher = GPTResearcher(
            query,
            **report_source="local,  # or hybrid**
            report_type=search_options['report_type'],
            tone=tone
        )
        
You can instead add the report source in the environment like this
**os.environ['REPORT_SOURCE'] ='local'**

it works, although it does not explain why the problem occurs

**Reply by ElishaKay** [reply]
Welcome @belviskhoremk 

Please upgrade to "0.11.0" and let us know if the issue is resolved for you

**Reply by belviskhoremk** [reply]
Yes the issue is resolved. Thank you very much

**Comment by ElishaKay** [comment]
Welcome @sankarsyneoshealth 

It's possible that some of the confusion here is that the researcher is still using web search as the first step to "generate the subquestions" of a given report.

Have a look at the example below at the section titled: "Pondering your question from several angles"

![local-docs-flow](https://github.com/user-attachments/assets/a45cca3a-70c4-47e0-a001-eda14bbb33e6)

In this case, the report source is set as "My Documents", but before the document content is analyzed, the researcher turns to the web to generate questions to guide it in exploring the documents thoroughly

I agree this is a confusing UX and something we should consider polishing.

Also, please upgrade to "0.11.0" and let us know if the issue is resolved for you.

**Reply by weipinglim** [reply]
I'm interest to run only run GPTResearcher in an environment without internet connection, solely to search within local docs. Wondering if there is a way to substitute or bypass this initial web search?

**Reply by ElishaKay** [reply]
Sup @weipinglim 

Try passing the Vector Store into the GPTResearcher Agent Class - [sample code here](https://docs.gptr.dev/docs/gpt-researcher/context/vector-stores)

We're moving in the direction of [making the Vector Store a swappable component](https://github.com/assafelovic/gpt-researcher/issues/1299), just like the Retriever and LLM are swappable components.

[Here's how data ingestion works under the hood](https://docs.gptr.dev/docs/gpt-researcher/context/data-ingestion)

To support your use case it sounds like the path of least resistance would be to first run a script to ingestion your data into a vector store and then [query the Vector Store directly](https://docs.gptr.dev/docs/gpt-researcher/context/vector-stores)
